crates/sui-types/src/messages.rs

```diff
@@ -1296,18 +1288,56 @@ impl TransactionData {
             gas_data: GasData {
                 price: DUMMY_GAS_PRICE,
                 owner: sender,
-                payment: gas_payment,
+                payment: vec![gas_payment],
                 budget: gas_budget,
             },
             expiration: TransactionExpiration::None,
         }
     }
+
+    pub fn new_system_transaction(kind: TransactionKind) -> Self {
+        // assert transaction kind if a system transaction?
+        // assert!(kind.is_system_tx());
+        let sender = SuiAddress::default();
+        TransactionData {
+            kind,
+            sender,
+            gas_data: GasData {
+                price: DUMMY_GAS_PRICE,
+                owner: sender,
+                payment: vec![(ObjectID::ZERO, SequenceNumber::default(), ObjectDigest::MIN)],
+                budget: 0,
+            },
+            expiration: TransactionExpiration::None,
+        }
+    }
+
     pub fn new(
         kind: TransactionKind,
         sender: SuiAddress,
         gas_payment: ObjectRef,
```

Comment:
Thoughts about just making this a vector? 
Too many spots to change?

Reply:
yep way toooooooo many, in time in time...